### PR TITLE
6 create class

### PR DESCRIPTION
**Pull request by heleena-fadi**

### CODE REVIEW

**Review by HamdyPs — CHANGES_REQUESTED** [review]
add check to prevent create same class name

**Comment by heleena-fadi** [comment]
it will be done

File: src/controllers/classes/addClass.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { type Request, type Response } from 'express';
+import Class from '../../database/classSchema';
+
+const addClass = async (req: Request, res: Response): Promise<void> => {
```
Comment (HamdyPs) [inline src/controllers/classes/addClass.ts]: add next function to be able to use custom error

File: src/controllers/classes/addClass.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  try {
+    const { className, description, trainerId, price, time, limit, userCount } = req.body;
+    const checkName = await Class.findOne({ className });
+    if (checkName) {
+      res.status(400).json({ msg: 'className already existed!' });
```
Comment (HamdyPs) [inline src/controllers/classes/addClass.ts]: use custom error

File: src/controllers/classes/addClass.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      description,
+      trainerId,
+      price,
+      time,
+      limit,
+      userCount,
+    });
+    res.status(201).json('class added successfully!');
+  } catch (error) {
+    console.log(error);
```
Comment (HamdyPs) [inline src/controllers/classes/addClass.ts]: remove console log and use custom error

File: src/database/classSchema.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  className: {
+    type: String,
+  },
+
+  description: {
+    type: String,
+  },
+
+  trainerId: {
+    type: Number,
```
Comment (HamdyPs) [inline src/database/classSchema.ts]: use mongoose type of object id>>>

File: src/interfaces/class.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+interface IClass {
+  className: string
+  description: string
+  trainerId: number
```
Comment (HamdyPs) [inline src/interfaces/class.ts]: use Type.objectId from mongoose not a number

File: src/validations/classesValidationSchema.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { object, string, number } from 'yup';
+
+export default object({
+  body: object({
+    className: string().required().min(3, 'must be at least 3 characters long'),
+    description: string().required(),
+    trainerId: string(),
+    price: number().required(),
+    time: number(),
```
Comment (HamdyPs) [inline src/validations/classesValidationSchema.ts]: add required

File: src/validations/classesValidationSchema.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+    limit: number(),
```
Comment (HamdyPs) [inline src/validations/classesValidationSchema.ts]: add required

File: src/validations/classesValidationSchema.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+    userCount: number(),
```
Comment (HamdyPs) [inline src/validations/classesValidationSchema.ts]: add required

**Review by HamdyPs — CHANGES_REQUESTED** [review]

File: src/validations/classesValidationSchema.ts (hunk 8 above)
Comment (heleena-fadi) [reply src/validations/classesValidationSchema.ts]: Thanks for useful comments , I hope always to give me a feedback like this .. I'm gonna fixing it

**Review by abeerAyad — APPROVED** [review]

**Review by AhmedMughessib — APPROVED** [review]

**Review by abeerAyad — APPROVED** [review]

**Review by HamdyPs — APPROVED** [review]

**Review by HamdyPs — APPROVED** [review]

**Review by abeerAyad — APPROVED** [review]

**Review by AdhamHaisami — APPROVED** [review]

**Pull request merged**